source/family/ambiq/ama3b1kk/target_reset.c

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file    target_reset.c
+ * @brief   Target reset for the new target
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2009-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//#include "target_reset.h"
+#include "util.h"
+#include "cmsis_os2.h"
+#include "target_config.h"
+#include "swd_host.h"
+#include "debug_cm.h"
+#include "DAP_config.h"
+#include "DAP.h"
+#include "target_family.h"
+#include "device.h"
+#include "daplink_debug.h"
+
+#define NVIC_Addr (0xe000e000)
+#define DBG_Addr (0xe000edf0)
+
+#define MCUCTRL_SCRATCH0 0x400401B0
+#define MCUCTRL_BOOTLDR 0x400401A0
+#define JDEC_PID 0xF0000FE0
+
+#if !defined(SCB_AIRCR_PRIGROUP_Pos)
+#define SCB_AIRCR_PRIGROUP_Pos 8U                              /*!< SCB AIRCR: PRIGROUP Position */
+#define SCB_AIRCR_PRIGROUP_Msk (7UL << SCB_AIRCR_PRIGROUP_Pos) /*!< SCB AIRCR: PRIGROUP Mask */
+#endif
+
+uint8_t swd_set_state_ama3b1kk(target_state_t state)
+{
+    uint32_t val, scratch0, secure; //jdecpid, bootldr
+    int8_t ap_retries = 2;
+    swd_init();
+    switch (state)
+    {
+    case RESET_HOLD:
+        debug_msg("swd_set_state_ama3b1kk RESET_HOLD \r\n");
+        swd_set_target_reset(1);
+        break;
+
+    case RESET_RUN:
+        debug_msg("swd_set_state_ama3b1kk RESET_RUN \r\n");
+        swd_set_target_reset(1);
+        osDelay(2);
+        swd_set_target_reset(0);
+        osDelay(2);
+        swd_off();
+        break;
+
+    case RESET_PROGRAM:
+        debug_msg("swd_set_state_ama3b1kk RESET_PROGRAM \r\n");
+        if (!swd_init_debug())
+        {
```

Comment:
These `{` should be attached, following the DAPLink coding style

Reply:
Latest commit changes parenthesis in this function to all start on same line, let me know if anything looks weird I am not  used to this style :D